src/inmanta/agent/executor.py

```diff
@@ -284,13 +326,32 @@ class VirtualEnvironmentManager:
     def __init__(self, envs_dir: str) -> None:
         self._environment_map: dict[EnvBlueprint, ExecutorVirtualEnvironment] = {}
         self.envs_dir: str = envs_dir
+        # We rely on a Named lock (`self._locks`) to be able to lock specific entries of the `_environment_map` dict. This
+        # allows us to prevent creating and deleting the same venv at a given time. The keys of this named lock are the hash of
+        # venv
         self._locks: NamedLock = NamedLock()
+        self._cleanup_scheduler = util.Scheduler("venv_cleanup_scheduler")
+
+    async def start(self) -> None:
+        interval = datetime.timedelta(days=1).total_seconds()
+        executor_venv_retention_time = cfg.executor_venv_retention_time.get()
+        assert datetime.timedelta(days=executor_venv_retention_time).total_seconds() > interval, (
+            "The `executor-venv-retention-time` should be larger than the period " "with which the executor touches its file!"
```

Comment:
I would also log the values

```suggestion
            "The `executor-venv-retention-time` should be larger then 1second"
```